vm/actor/tests/deadline_test.rs

```diff
@@ -0,0 +1,653 @@
+    assert_eq!(1, partitions.len());
+
+    assert_bf_equal(&bf_seq(1234, PART_SIZE), partitions.get(0).unwrap());
+}
+
+#[test]
+fn full_plus_partial_test() {
+    let mut dls = Deadlines::new();
+    dls.due[10] = bf_seq(5555, PART_SIZE + 1);
+
+    let mut partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 10, &[0]).unwrap();
+    assert_eq!(1, partitions.len());
+    assert_bf_equal(&bf_seq(5555, PART_SIZE), partitions.get(0).unwrap());
+
+    partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 10, &[1]).unwrap();
+    assert_eq!(1, partitions.len());
+    assert_bf_equal(&bf_seq(5555 + PART_SIZE, 1), partitions.get(0).unwrap());
+
+    partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 10, &[0, 1]).unwrap();
+    assert_eq!(2, partitions.len());
+    assert_bf_equal(&bf_seq(5555, PART_SIZE), partitions.get(0).unwrap());
+    assert_bf_equal(&bf_seq(5555 + PART_SIZE, 1), partitions.get(1).unwrap());
+}
+
+#[test]
+fn multiple_partition_test() {
+    let mut dls = Deadlines::new();
+    dls.due[1] = bf_seq(0, 3 * PART_SIZE + 1);
+
+    let partitions =
+        compute_partitions_sector(&mut dls, PART_SIZE as u64, 1, &[0, 1, 2, 3]).unwrap();
+    assert_eq!(4, partitions.len());
+
+    assert_bf_equal(&bf_seq(0, PART_SIZE), partitions.get(0).unwrap());
+    assert_bf_equal(
+        &bf_seq(1 * PART_SIZE, PART_SIZE),
+        partitions.get(1).unwrap(),
+    );
+    assert_bf_equal(
+        &bf_seq(2 * PART_SIZE, PART_SIZE),
+        partitions.get(2).unwrap(),
+    );
+    assert_bf_equal(&bf_seq(3 * PART_SIZE, 1), partitions.get(3).unwrap());
+}
+
+#[test]
+fn numbered_partitions_test() {
+    let mut dls = Deadlines::new();
+    dls.due[1] = bf_seq(0, 3 * PART_SIZE + 1);
+    dls.due[3] = bf_seq(3 * PART_SIZE + 1, 1);
+    dls.due[5] = bf_seq(3 * PART_SIZE + 2, 2 * PART_SIZE);
+
+    let mut partitions =
+        compute_partitions_sector(&mut dls, PART_SIZE as u64, 1, &[0, 1, 2, 3]).unwrap();
+    assert_eq!(4, partitions.len());
+
+    partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 3, &[4]).unwrap();
+    assert_eq!(1, partitions.len());
+    assert_bf_equal(&bf_seq(3 * PART_SIZE + 1, 1), partitions.get(0).unwrap());
+
+    partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 5, &[5, 6]).unwrap();
+    assert_eq!(2, partitions.len());
+    assert_bf_equal(
+        &bf_seq(3 * PART_SIZE + 2, PART_SIZE),
+        partitions.get(0).unwrap(),
+    );
+    assert_bf_equal(
+        &bf_seq(3 * PART_SIZE + 2 + PART_SIZE, PART_SIZE),
+        partitions.get(1).unwrap(),
+    );
+}
+
+#[test]
+#[should_panic]
+fn numbered_partitions_should_err_test() {
+    let mut dls = Deadlines::new();
+    dls.due[1] = bf_seq(0, 3 * PART_SIZE + 1);
+    dls.due[3] = bf_seq(3 * PART_SIZE + 1, 1);
+    dls.due[5] = bf_seq(3 * PART_SIZE + 2, 2 * PART_SIZE);
+
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 1, &[4]).unwrap();
```

Comment:
wouldnt this just panic on the first one and end the test?

Reply:
yeah just assert each are an error or none (I missed the should panic)